ledger-api/grpc-definitions/com/daml/ledger/api/v1/event.proto

```diff
@@ -82,6 +82,23 @@ message CreatedEvent {
   // This means a newer client works with an older sandbox seamlessly.
   // Optional
   google.protobuf.StringValue agreement_text = 6;
+
+  // Metadata of the contract. Required for contracts created
+  // after the introduction of this feature.
+  ContractMetadata metadata = 10;
+}
+
+// Contract metadata which can be used in DisclosedContract.
+message ContractMetadata {
+  // Ledger effective time of the transaction that created the contract.
+  // Required
+  google.protobuf.Timestamp created_at = 1;
+  // Hash of the contract key if defined.
+  // Optional
+  bytestring contract_key_hash = 2;
+  // Driver-specific metadata. This is opaque and cannot be decoded.
+  // Required
+  bytestring driver_metadata = 3;
```

Comment:
Do you have any reason for not using the Protobuf primitive `bytes` type?
```suggestion
  bytes driver_metadata = 3;
```

Reply:
good catch fixed